src/BenchmarkDotNet/Validators/CompilationValidator.cs

```diff
@@ -25,8 +26,38 @@ private CompilationValidator() { }
         public IEnumerable<ValidationError> Validate(ValidationParameters validationParameters)
             => ValidateCSharpNaming(validationParameters.Benchmarks)
                     .Union(ValidateNamingConflicts(validationParameters.Benchmarks))
+                    .Union(ValidateClassModifiers((validationParameters.Benchmarks))
                     .Union(ValidateAccessModifiers(validationParameters.Benchmarks))
-                    .Union(ValidateBindingModifiers(validationParameters.Benchmarks));
+                    .Union(ValidateBindingModifiers(validationParameters.Benchmarks))
+                );
+
+        private static IEnumerable<ValidationError> ValidateClassModifiers(IEnumerable<BenchmarkCase> benchmarks)
+        {
+            return benchmarks
+                .Distinct(BenchmarkMethodEqualityComparer.Instance)
+                .SelectMany(benchmark =>
+                {
+                    var errors = new List<ValidationError>();
+                    string message = string.Empty;
+                    if (benchmark.Descriptor.Type.IsSealed){ message = "a sealed class"; }
+                    if (!benchmark.Descriptor.Type.IsPublic && !benchmark.Descriptor.Type.IsNestedPublic) { message = "a private class"; }
```

Comment:
You can add both to the list.

Reply:
I did not understand this

Reply:
You're overwriting the string message and adding only a single validation error. Instead, you can add the validation error in each if-check.